### PR TITLE
Graphs

### PR DESCRIPTION
**Pull request by PijusOz**

### CODE REVIEW

File: back-end-side/Controllers/Sorting.cs [hunk 1]
```diff
@@ -52,13 +77,51 @@ public SortingModel SortToCategories(string pickedDate = "")
...
+
+                Model.PreviousYearFoodSum = queryMethod("food", Supermarkets, previousYear);
+                Model.PreviousYearClothesSum = queryMethod("clothes", ClothesShops, previousYear);
+                Model.PreviousYearCarSum = queryMethod("car", CarMaintenanceShops, previousYear);
+                Model.PreviousYearEntertaintmentSum = queryMethod("entertainment", EntertainmentShops, previousYear);
+                Model.PreviousYearHouseSum = queryMethod("house", HouseMaintenanceShops, previousYear);
 
                 double sumOfOther = 0;
+                double sumOfPreviousOther = 0;
+                double sumOfPreviousYearOther = 0;
```
Comment (dissco6) [inline back-end-side/Controllers/Sorting.cs:124]: Isn't ```previousYear``` missing right here?

File: back-end-side/Controllers/Sorting.cs [hunk 2]
```diff
@@ -81,13 +155,14 @@ public SortingModel SortToCategories(string pickedDate = "")
                 }
 
                 Model.OtherSum = sumOfOther;
+                Model.PreviousOtherSum = sumOfPreviousOther;
```
Comment (dissco6) [inline back-end-side/Controllers/Sorting.cs:158]: Isn't ```previousYear``` missing right here?

Comment (PijusOz) [reply back-end-side/Controllers/Sorting.cs:158]: oh-oh

File: front-end-side/src/Pages/Data.js [hunk 3]
```diff
@@ -121,154 +126,239 @@ const EnterData = () => {
...
-                                                <ExpensesCard name="transportation" categorySum={categoryData.data.carSum} expenses={newExpenses} />
-                                                <ExpensesCard name="entertainment" categorySum={categoryData.data.entertaintmentSum} expenses={newExpenses} />
-                                                <ExpensesCard name="house" categorySum={categoryData.data.houseSum} expenses={newExpenses} />
-                                                <ExpensesCard name="clothes" categorySum={categoryData.data.clothesSum} expenses={newExpenses} />
-                                                <ExpensesCard name="other" categorySum={categoryData.data.otherSum} expenses={newExpenses} />
+                                                <div id="statistics-graph">
+                                                    <div class="graph-block">
+                                                        <div className="title">{toMonthName(datePick.slice(5, 7))} compared with {toMonthName(datePick.slice(5, 7) - 1)}</div>
```
Comment (dissco6) [inline front-end-side/src/Pages/Data.js:311]: What if user have selected his total expenses? With what graphs are comparing total expenses?

File: back-end-side/Controllers/Sorting.cs [hunk 4]
```diff
@@ -52,13 +77,51 @@ public SortingModel SortToCategories(string pickedDate = "")
...
+                    previousMonth = dateFromFront.AddMonths(-1);
+                }
+                else
+                {
+                    dateFromFront = DateTime.ParseExact(pickedDate, "yyyy-MM", null);
+                    previousYear = dateFromFront.AddYears(-1);
+                    previousMonth = dateFromFront.AddMonths(-1);
+                }
+
+                Model.FoodSum = queryMethod("food", Supermarkets, dateFromFront);
```
Comment (dissco6) [inline back-end-side/Controllers/Sorting.cs:104]: Could we make those less repeated?

Comment (PijusOz) [reply back-end-side/Controllers/Sorting.cs:104]: Nope

File: front-end-side/src/Pages/Data.js (hunk 3 above)
Comment (PijusOz) [reply front-end-side/src/Pages/Data.js:311]: Good idea. It will be changed in the future.

**Pull request merged**